File: easy/max-consecutive-ones.py
class Solution:
    def findMaxConsecutiveOnes(self, nums: List[int]) -> int:
        s = nums
        n = len(s)

        m1 = 0
        m0 = 0

        t = True if s[0] == 1 else False  # 1 , 0 -> True
        start = 0
        end = 0

        while end < n:
            if t and s[end] == 1:
                end += 1

            elif not t and s[end] == 0:
                end += 1

            else:
                dist = end - start
                if t:
                    if dist > m1:
                        m1 = dist
                else:
                    if dist > m0:
                        m0 = dist

                start = end
                t = not t
        dist = end - start
        if t:
            if dist > m1:
                m1 = dist
        else:
            if dist > m0:
                m0 = dist

        return m1

File: easy/max-consecutive-ones.cpp
class Solution {
public:
    int findMaxConsecutiveOnes(vector<int>& nums) {
        int n = nums.size();
        
        for (int i =0; i < n; i ++){
            int start = i;
            
            int count = 0;
            while (i < n && nums[i] == 1){
                count +=1;
                nums[i] = -1;
                i ++;
            }
            nums[start] = count; 
            if (nums[start] > 0){
                nums[start+count -1] = count;
            }
        }
        
        int _max = 0;
        for (int i =0; i < n; i++)
        {
            _max = max(nums[i], _max);
        }
        
        return _max;
    };
};
